sdks/go/pkg/beam/core/runtime/exec/datasource_test.go

```diff
@@ -494,11 +501,50 @@ func TestSplitHelper(t *testing.T) {
 		for _, test := range tests {
 			test := test
 			t.Run(fmt.Sprintf("(%v of [%v, %v])", test.frac, test.curr, test.size), func(t *testing.T) {
-				got, err := splitHelper(test.curr, test.size, nil, test.frac)
+				wantFrac := 0.0
+				got, gotFrac, err := splitHelper(test.curr, test.size, 0.0, nil, test.frac, false)
+				if err != nil {
+					t.Errorf("error in splitHelper: %v", err)
+				} else {
+					if got != test.want {
+						t.Errorf("incorrect split point: got: %v, want: %v", got, test.want)
+					}
+					if !floatEquals(gotFrac, wantFrac, eps) {
+						t.Errorf("incorrect split fraction: got: %v, want: %v", gotFrac, wantFrac)
+					}
+				}
+			})
+		}
+	})
+
+	t.Run("WithElementProgress", func(t *testing.T) {
+		tests := []struct {
+			curr, size int64
+			currProg   float64
+			frac       float64
+			want       int64
+		}{
+			// Progress into the active element influences where the split of
+			// the remainder falls.
+			{curr: 0, currProg: 0.5, size: 4, frac: 0.25, want: 1},
+			{curr: 0, currProg: 0.9, size: 4, frac: 0.25, want: 2},
+			{curr: 1, currProg: 0.0, size: 4, frac: 0.25, want: 2},
+			{curr: 1, currProg: 0.1, size: 4, frac: 0.25, want: 2},
+		}
+		for _, test := range tests {
+			test := test
+			t.Run(fmt.Sprintf("(%v of [%v, %v])", test.frac, float64(test.curr)+test.currProg, test.size), func(t *testing.T) {
+				wantFrac := 0.0
+				got, gotFrac, err := splitHelper(test.curr, test.size, test.currProg, nil, test.frac, false)
 				if err != nil {
 					t.Errorf("error in splitHelper: %v", err)
-				} else if got != test.want {
-					t.Errorf("incorrect split point: got: %v, want: %v", got, test.want)
+				} else {
```

Comment:
Same else comment here. In this case, use t.Fatalf in the err != nil case instead of t.Errorf to get the desired behaviour.

Reply:
Done.